File: ios/DietApp/DietApp/Models/Message.swift
import Foundation

struct Message: Codable, Identifiable {
    let id: UUID
    let senderId: UUID
    let receiverId: UUID
    let content: String?
    let imageUrl: String?
    let isRead: Bool
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id, content
        case senderId = "sender_id"
        case receiverId = "receiver_id"
        case imageUrl = "image_url"
        case isRead = "is_read"
        case createdAt = "created_at"
    }

    var createdDate: Date? {
        let formatter = ISO8601DateFormatter()
        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        if let date = formatter.date(from: createdAt) { return date }
        formatter.formatOptions = [.withInternetDateTime]
        if let date = formatter.date(from: createdAt) { return date }
        let df = DateFormatter()
        df.locale = Locale(identifier: "en_US_POSIX")
        df.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS"
        if let date = df.date(from: createdAt) { return date }
        df.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
        return df.date(from: createdAt)
    }
}

struct Conversation: Codable, Identifiable {
    let userId: UUID
    let fullName: String
    let lastMessage: String?
    let lastMessageAt: String?
    let unreadCount: Int

    var id: UUID { userId }

    enum CodingKeys: String, CodingKey {
        case fullName = "full_name"
        case userId = "user_id"
        case lastMessage = "last_message"
        case lastMessageAt = "last_message_at"
        case unreadCount = "unread_count"
    }
}

struct MessageCreate: Codable {
    let receiverId: UUID
    let content: String?
    let imageUrl: String?

    enum CodingKeys: String, CodingKey {
        case receiverId = "receiver_id"
        case content
        case imageUrl = "image_url"
    }
}

File: backend/app/schemas/message.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, model_validator


class MessageCreate(BaseModel):
    receiver_id: UUID
    content: str | None = None
    image_url: str | None = None

    @model_validator(mode="after")
    def check_content_or_image(self):
        if not self.content and not self.image_url:
            raise ValueError("content veya image_url gerekli")
        return self


class MessageResponse(BaseModel):
    id: UUID
    sender_id: UUID
    receiver_id: UUID
    content: str | None
    image_url: str | None
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class ConversationResponse(BaseModel):
    user_id: UUID
    full_name: str
    last_message: str | None
    last_message_at: datetime
    unread_count: int
